app/models/concerns/orchestration/ssh_provision.rb

```diff
@@ -82,6 +82,10 @@ def setSSHProvision
       Host.find(id).built
       respond_to?(:initialize_puppetca,true) && initialize_puppetca && delAutosign if puppetca?
     else
+      if Setting[:clean_up_failed_deployment]
+        logger.info "Deleting host #{name} because of non zero exit code of deployment script."
```

Comment:
Nitpick, if you could include the name of the script that'd help users when they debug what's going on.

Aside from that, make sure to include the `raise ::Foreman::Exception` (line 89) within this else block, otherwise it's going to raise the exception in both cases (success and error)